integration/vm_factory/vm_templating_test.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 HyperHQ Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This test will perform several tests to validate kata containers with
+# vm templating.
+
+set -e -x
+
+cidir=$(dirname "$0")
+
+source "${cidir}/../../metrics/lib/common.bash"
+
+# Environment variables
+IMAGE="${IMAGE:-busybox}"
+CONTAINER_NAME=$(random_name)
+PAYLOAD_ARGS="${PAYLOAD_ARGS:-tail -f /dev/null}"
+
+if [ -f ${DEFAULTCONFIG} ]; then
+	runtime_config_path=${DEFAULTCONFIG}
```

Comment:
I wonder if we are better finding these via a `kata-runtime kata-env --json | jq blah-blah` or similar `kata-env` based query? Similarly, we could use `kata-env` to find the actual hypervisor path and replace the `ps | grep qemu` elsewhere in the file with a `pgrep $HYPERVISOR_PATH` type thing.

Reply:
It is not just for querying. I actually need to modify the config file and change it back in the test. See `enable/disable_vm_template_config`. Can I do that with `kata-runtime kata-env`?

Reply:
Hi. You cannot modify using `kata-runtime kata-env`, but you can use `kata-env` to find the actual config file path the runtime is using, and then modify that with `sed` like you are already.

In fact, we already have a piece of code in the metrics common.lib that does the extraction https://github.com/kata-containers/tests/blob/master/metrics/lib/common.bash#L34
```
	RUNTIME_CONFIG_PATH=$(awk '/^\[Runtime\]$/ {foundit=1} /^    Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
```
(this was written before we had the json kata-env, which might be slightly cleaner ;-) )
That code is called from the metrics `common_init()`, which you are probably not calling. You could call directly `extract_kata_env` to get RUNTIME_CONFIG_PATH set in the env.
I think using `kata-env` to find the config file the runtime is actually using is probably the right thing to do - having the paths wired into `common.bash` feels wrong.

Reply:
Thanks @grahamwhaley. I'll update the PR accordingly.